src/main/java/org/apache/solr/handler/dataimport/CryptoKeys.java

```diff
@@ -0,0 +1,163 @@
+package org.apache.solr.handler.dataimport;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import java.nio.charset.Charset;
+import java.security.GeneralSecurityException;
+import java.security.MessageDigest;
+import java.util.Arrays;
+import java.util.Base64;
+
+import org.apache.solr.common.SolrException;
+
+/**
+ * Code copied from Solr 8.11.1
+ */
+public class CryptoKeys {
```

Comment:
note: `decodeAES` method is used in `JdbcDataSource` class. It was removed from Solr 9, so I copied it here from 8.11.1.